packages/payload/src/queues/localAPI.ts

```diff
@@ -71,11 +73,24 @@ export const getJobsLocalAPI = (payload: Payload) => ({
       data.taskSlug = args.task as string
     }
 
-    return (await payload.create({
-      collection: 'payload-jobs',
-      data,
-      req: args.req,
-    })) as TTaskOrWorkflowSlug extends keyof TypedJobs['workflows']
+    const resultingDoc =
+      payload?.config?.jobs?.depth || payload?.config?.jobs?.runHooks
+        ? await payload.create({
+            collection: 'payload-jobs',
+            data,
+            depth: payload.config.jobs.depth ?? 0,
+            req: args.req,
+          })
+        : jobAfterRead({
+            config: payload.config,
+            doc: await payload.db.create({
+              collection: 'payload-jobs',
+              data,
+              req: args.req,
+            }),
+          })
+
+    return resultingDoc as TTaskOrWorkflowSlug extends keyof TypedJobs['workflows']
```

Comment:
Feelsl like it would be a lot easier to read this if you used if statements instead of a ternary and instead of assigning resultingDoc to return it just return the the different calls.
```ts
if (  payload?.config?.jobs?.depth || payload?.config?.jobs?.runHooks) {
	return await payload.create({
    collection: 'payload-jobs',
    data,
    depth: payload.config.jobs.depth ?? 0,
    req: args.req,
  })
} 
return jobAfterRead({
  config: payload.config,
  doc: await payload.db.create({
    collection: 'payload-jobs',
    data,
    req: args.req,
  }),
})
```